internal/provider/provider.go

```diff
@@ -61,6 +69,18 @@ func (p *Provider) Schema(ctx context.Context, req provider.SchemaRequest, resp
 	}
 }
 
+// combineMaps combines two maps, with the first map (left) taking precedence.
+func combineMaps(left, right map[string]string) map[string]string {
```

Comment:
I feel like the fact that we are combining things makes `extra_` a more appropriate prefix, but I can also see where you are coming from with `default_` because the provider-level one won't be added if the same key is present in the config file (more like a default for that key).

I don't feel super strongly either way.